proposed/dotnet-bootstrapper-muxer-path.md

```diff
@@ -0,0 +1,68 @@
+
+This bootstrapper would be acquired in the same fashion as `nvm`:
+Via a website with a simple install script that downloads the executable.
+
+The .NET SDK would also shell out to the bootstrapper executable for commands it does not know, aka `dotnet install`, `dotnet uninstall`, etc. This way the bootstrapper could automatically update itself, say for example when the `releases.json` url changed, this could be a breaking change requiring a self-update to function.
+
+From a security context, it may be better to bundle this alongside the host in the same directory, for this would share the same security context.
+
+In theory, this is 'doable' for .NET 10 in terms of changing the muxer to know about a specified bootstrapper location, which would enable these scenarios outside of just .NET 11.
+
+
+### Create a C# Muxer
+
+The muxer would become a bootstrapper and the muxer itself. It would cease to become C++ and become ported C# code that is run as native AOT code. This is high-risk and also likely still lower performance than a C++ application.
+
+
+## PATH Issue
+
+Visual Studio and the MSI or 'global' .NET Installer set the System `PATH` on Windows and not the user level `PATH`. The system `PATH` always takes precedence over the user `PATH`. If `dnvm` sets the user path to enable a folder of local .NET SDKs to be discovered by the `dotnet.exe` application host at `DOTNET_ROOT`, via the `PATH`, those installs would become 'undetectable' to the runtime, debugger, etc when Visual Studio would update and modify the System `PATH`.
+
+This is also applicable to `source-built` SDKs and the `pkg` installer.
+
+There have been several methods suggested to handle this. None of them are ideal.
+
+### Fail on Windows if System PATH set, if `paths` in `global.json` is not set
+
+In this approach, we prioritized defined and replicable behavior over the easier UX behavior. System administrator installs are created in predictable locations, especially those from Visual Studio and/or package manager feeds.
+
+What we could do is look for those feeds per install operation and decline to do anything when a user tries to run `dotnet install`. The error message would point the user towards leveraging the `sdk: paths []` option in their repository's `global.json` file, which would specify to the host or `dotnet.exe` where to look for the installs of the .NET SDK or .NET runtimes. The host is already aware of this as of .NET 10.
+
+This is subject to backlash, especially as only 2 to 6% of users are on a `global.json` according to internal metrics.
```

Comment:
```suggestion
This is subject to negative feedback, especially as only 2 to 6% of users are on a `global.json` according to internal metrics.
```

Reply:
I rewrote this to focus on the bad experience / work it creates for users, rather than the negative feedback. Thank you.